### PR TITLE
Support --lr-scale-sqrt and tidy up associated output

### PR DESCRIPTION
**Pull request by clayne**
* By default, `--lr-scale` is the product of learning rate, batch size, gradient accumulation, and number of GPUs. This adds an additional option `--lr-scale-sqrt` to instead multiply by the square root of (batch size, gradient accumulation, and number of GPUs). This is being provided as an option as there are multiple schools of thought on what is "correct" - so let the user decide.

* Related log output cleaned up a bit to display scientific notation and emit useful info at the end.

Example output:

`--lr-scale`:
```
Scaling learning rate from 1.0e-04 to 8.0e-04 due to --lr-scale (bsz: 8, ga: 1, nprocs: 1)
```
`--lr-scale` and `--lr-scale-sqrt`:
```
Scaling learning rate from 1.0e-04 to 2.8e-04 due to --lr-scale and --lr-scale-sqrt (bsz: 8, ga: 1, nprocs: 1)
```

### CODE REVIEW

**Comment by bghira** [comment]
i've redirected it toward v2 so there's fewer conflicts for that set of changes.

**Comment by clayne** [comment]
> i've redirected it toward v2 so there's fewer conflicts for that set of changes.

I think you might have to rebase your refactor branch to pull in changes from `main` - otherwise I could probably do it (reorient it for the refactor branch) on my end if needed.

**Review by bghira — APPROVED** [review]
